registration/models.py

```diff
@@ -376,3 +377,364 @@ def send_activation_email(self, site, request=None):
+
+        if getattr(settings, 'REGISTRATION_EMAIL_HTML', True):
+            try:
+                message_html = render_to_string(
+                    admin_approve_email_html, ctx_dict)
+            except TemplateDoesNotExist:
+                pass
+            else:
+                email_message.attach_alternative(message_html, 'text/html')
+
+        email_message.send()
+
+    def delete_expired_users(self):
+        """
+        Remove expired instances of ``SupervisedRegistrationProfile`` and their
+        associated ``User``s.
+
+        Accounts to be deleted are identified by searching for
+        instances of ``RegistrationProfile`` with expired activation
+        keys, and then checking to see if their associated ``User``
+        instances have the field ``is_active`` set to ``False``; any
+        ``User`` who is both inactive and has an expired activation
+        key will be deleted.
+
+        It is recommended that this method be executed regularly as
+        part of your routine site maintenance; this application
+        provides a custom management command which will call this
+        method, accessible as ``manage.py cleanupregistration``.
+
+        Regularly clearing out accounts which have never been
+        activated serves two useful purposes:
+
+        1. It alleviates the ocasional need to reset a
+           ``SupervisedRegistrationProfile`` and/or re-send an activation email
+           when a user does not receive or does not act upon the
+           initial activation email; since the account will be
+           deleted, the user will be able to simply re-register and
+           receive a new activation key.
+
+        2. It prevents the possibility of a malicious user registering
+           one or more accounts and never activating them (thus
+           denying the use of those usernames to anyone else); since
+           those accounts will be deleted, the usernames will become
+           available for use again.
+
+        If you have a troublesome ``User`` and wish to disable their
+        account while keeping it in the database, simply delete the
+        associated ``SupervisedRegistrationProfile``; an inactive ``User`` which
+        does not have an associated ``RegistrationProfile`` will not
+        be deleted.
+
+        """
+        for profile in self.all():
+            try:
+                if profile.activation_key_expired():
+                    user = profile.user
+                    if not user.is_active:
+                        profile.delete()
+                        user.delete()
+            except UserModel().DoesNotExist:
+                profile.delete()
+
+
+@python_2_unicode_compatible
+class SupervisedRegistrationProfile(RegistrationProfile):
+
+    # Same model as ``RegistrationProfile``, just a different
+    # Manager to implement the extra functionality required
+    # in admin approval
+    objects = SupervisedRegistrationManager()
+
+    def __str__(self):
```

Comment:
I guess not, since if I remove the declaration I have not explicitly defined `__str__() `. https://docs.djangoproject.com/en/dev/ref/utils/#django.utils.encoding.python_2_unicode_compatible
I guess inheriting this from the superclass is not enough.